utils/ScopedIncrement.h

```diff
@@ -10,12 +11,20 @@ class ScopedIncrement
 public:
     explicit ScopedIncrement(std::atomic_uint32_t& counter) : _counter(counter) { ++_counter; }
 
-    ~ScopedIncrement() { --_counter; }
+    ~ScopedIncrement()
+    {
+#if DEBUG
+        auto value = --_counter;
+        assert(value != 0xFFFFFFFFu);
+#else
+        --counter;
+#endif
+    }
 
     std::atomic_uint32_t& getCounter() { return _counter; }
```

Comment:
This looks fishy since it allows to modify _counter. I have not found usage of it. If what is needed is just a check of counter's value  - consider returning "const uint32" and mark method as "const".

Reply:
Disregard. I got what you wanted to do (read a comment in job.h)